src/imageops/colorops.rs

```diff
@@ -181,14 +179,51 @@ where
 pub trait ColorMap {
     /// The color type on which the map operates on
     type Color;
-    /// Returns the index of the closed match of `color`
+    /// Returns the index of the closest match of `color`
     /// in the color map.
     fn index_of(&self, color: &Self::Color) -> usize;
+    /// Looks up color by index in the color map.  If `idx` is out of range for the color map, or
+    /// ColorMap doesn't implement `lookup` `None` is returned.
+    fn lookup(&self, _idx: usize) -> Option<Self::Color> {
```

Comment:
This name is fine, nevertheless methods of the form `fn (&Self, usize) -> Option<_>` are often named `get`. However, that has a larger risk associated with it by colliding with the standard name (see considerations in the rfc). Another idea would be to name it `get_color` which should be much less risky. At the same time though, the name association to `has_lookup` is nice, we would have to find another name for that method then.

I don't want to spend too much energy on bikeshedding the name here, it's already pretty good. So, if there is no immediate idea for another pair of names then this is perfectly fine.

Reply:
I didn't notice this concern on the first review either, sorry for any delay.

Reply:
I can do `get_color`/`has_get_color` if that seems better, or I'm okay leaving it as-is?  I have no strong preference, and I'm happy to let you decide on the style as the library maintainer.

Reading https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter I'm not sure `get` would be a good name, as I think there are other intuitive color map methods that could be named `get`, i.e. "return the underlying palette for the color map.  They point out `first` as an example where it is not named `get_first`, so I think `lookup` can be considered sufficient.  In less functional language `map` would have also been a good name, but that's a terrible name in rust due to the pervasive use of `map` for functional-style programming.

Reply:
Let's leave it as is then, those are subtle differences that shouldn't be surprises.